File: 167_two_sum_sorted/two_sum.ts
function twoSum(numbers: number[], target: number): number[] {
  let [l, r] = [0, numbers.length - 1];

  while (l < r) {
    if (numbers[l] + numbers[r] == target) {
      return [l + 1, r + 1];
    } else if (numbers[l] + numbers[r] > target) {
      r--;
    } else if (numbers[l] + numbers[r] < target) {
      l++;
    }
  }
  return [];
}

File: 167_two_sum_sorted/TwoSum.java
class TwoSum {
     public int[] twoSum(int[] numbers, int target) {
        int l = 0;
        int r = numbers.length - 1;

        while (l < r) {
            if (numbers[l] + numbers[r] == target) {
                return new int[] {l + 1, r + 1};
            }

             if (numbers[l] + numbers[r] > target) {
                r--;
            }

             if (numbers[l] + numbers[r] < target) {
                l++;
            }
        }

         return new int[] {l + 1, r + 1};
    }
}